convert_hf_to_gguf.py

```diff
@@ -1989,6 +1989,15 @@ def set_vocab(self):
         except FileNotFoundError:
             self._set_vocab_gpt2()
 
+    def set_gguf_parameters(self):
+        super().set_gguf_parameters()
+        if self.hparams.get("rope_scaling") is not None and "factor" in self.hparams["rope_scaling"]:
+            if self.hparams["rope_scaling"].get("type") == "yarn":
+                self.gguf_writer.add_rope_scaling_type(gguf.RopeScalingType.YARN)
+                self.gguf_writer.add_rope_scaling_factor(self.hparams["rope_scaling"]["factor"])
+                self.gguf_writer.add_rope_scaling_orig_ctx_len(self.hparams["rope_scaling"]["original_max_position_embeddings"])
+                self.gguf_writer.add_rope_scaling_yarn_log_mul(0.1)
```

Comment:
Where is this value derived from?

Reply:
this is in my original comment in the discussion. line 1999 is 0.1 * 1.0 (in place of 0.707 as used in the reference model). The other values are what are required to set yarn by way of a scaling factor

Reply:
The value ends up not being used. If you search for `LLM_KV_ROPE_SCALING_YARN_LOG_MUL`, the only hit is for the DeepSeek arch:

https://github.com/ggerganov/llama.cpp/blob/47844dc2327313fe841f6688bba8c058df8fc7b7/src/llama.cpp#L6072

Reply:
hmm, the _default_ value should be 1.0, may be why it still works.. looking to see if I can spot that.

Reply:
I think, if I understand this right, I should remove that line, because it should have been 1.0 anyway and it is not being used. Everything else is correct, for ARCH QWEN2.

But .. the implementation for yarn could be improved for DeepSeek. Maybe we could leave this open so someone can ensure we don't want to change rope_yarn (and upstream methods) in a few places at the same time.

Reply:
Thanks for taking a detailed look. Seems we are good to merge this.